### PR TITLE
get System menu's IOS instead of assuming 80

### PR DESCRIPTION
**Pull request by Naim2000**

### CODE REVIEW

File: patcher.cpp [hunk 1]
```diff
@@ -26,6 +26,22 @@ static s32 GetWC24Error(s32 wc24_fd, NWC24Config& config)
   return reinterpret_cast<s32 *>(io_buf)[2];
 }
 
+static s32 GetSystemMenuIOS() {
+  s32 ret;
+  u32 viewSize = 0;
+
+  ret = ES_GetTMDViewSize(0x100000002LL, &viewSize);
+  if (ret < 0) return ret;
```
Comment (noahpistilli) [inline patcher.cpp]: New line for the return

File: patcher.cpp [hunk 2]
```diff
@@ -26,6 +26,22 @@ static s32 GetWC24Error(s32 wc24_fd, NWC24Config& config)
...
+
+  u8 _buffer[viewSize] = {};
+  tmd_view* tmd = reinterpret_cast<tmd_view*> (_buffer);
+  ret = ES_GetTMDView(0x100000002LL, _buffer, viewSize);
+  if (ret < 0) return ret;
```
Comment (noahpistilli) [inline patcher.cpp]: Ditto.

File: patcher.cpp [hunk 3]
```diff
@@ -26,6 +26,22 @@ static s32 GetWC24Error(s32 wc24_fd, NWC24Config& config)
...
 }
 
+static s32 GetSystemMenuIOS() {
+  s32 ret;
+  u32 viewSize = 0;
+
+  ret = ES_GetTMDViewSize(0x100000002LL, &viewSize);
+  if (ret < 0) return ret;
+
+  u8 _buffer[viewSize] = {};
```
Comment (noahpistilli) [inline patcher.cpp]: Consider `std::array<u8, view_size> _buffer{}`

File: patcher.cpp [hunk 4]
```diff
@@ -26,6 +26,22 @@ static s32 GetWC24Error(s32 wc24_fd, NWC24Config& config)
   return reinterpret_cast<s32 *>(io_buf)[2];
 }
 
+static s32 GetSystemMenuIOS() {
+  s32 ret;
+  u32 viewSize = 0;
```
Comment (noahpistilli) [inline patcher.cpp]: `u32 view_size{};`

**Review by noahpistilli — CHANGES_REQUESTED** [review]
Also don't put spaces after the casts.

**Comment by Naim2000** [comment]
Am I cool now or do I ssh to my laptop and squash everything

**Comment by noahpistilli** [comment]
You may need to include the array header for it to compile, otherwise looks good to squash

**Comment by Naim2000** [comment]
That came out much worse than I expected. Memalign time(?)
Screenshot_20231014-214356

**Comment by noahpistilli** [comment]
include the array header and pass `_buffer.data()` to `ES_GetTMDView`

**Comment by Naim2000** [comment]
:( I think it's memalign time
Screenshot_20231014-215219


Also did I outright manage to discard the alignas(0x20)? Swear I had that

**Comment by Naim2000** [comment]
Look the c++ syntax stuff is cool but it doesn't want to work out right now. Values in angle brackets must be known at compile time

File: patcher.cpp [hunk 5]
```diff
@@ -26,6 +27,24 @@ static s32 GetWC24Error(s32 wc24_fd, NWC24Config& config)
...
 
+static s32 GetSystemMenuIOS() {
+  s32 ret;
+  u32 view_size{};
+
+  ret = ES_GetTMDViewSize(0x100000002LL, &view_size);
+  if (ret < 0)
+    return ret;
+
+  std::array<u8, view_size> _buffer{};
```
Comment (noahpistilli) [inline patcher.cpp]: You will need to use a vector instead, array doesn't like that `view_size` is not const

```cpp
  std::vector<u8> _buffer(view_size);
  tmd_view* tmd = reinterpret_cast<tmd_view*>(_buffer.data());
```

**Comment by Naim2000** [comment]
Good to go. Also systemd slept on me in my ssh so i had to get out of bed :/
image

**Review by noahpistilli — APPROVED** [review]

**Pull request merged**